atlasdb-timelock-server/src/test/java/com/palantir/atlasdb/timelock/ServerImplementationTest.java

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright 2017 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.timelock;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.jetty.util.component.LifeCycle;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.palantir.atlasdb.timelock.config.TimeLockServerConfiguration;
+
+import io.dropwizard.jersey.setup.JerseyEnvironment;
+import io.dropwizard.lifecycle.setup.LifecycleEnvironment;
+import io.dropwizard.setup.Environment;
+
+public abstract class ServerImplementationTest {
```

Comment:
This class was actually supposed to be the `AbstractServerImplementationTest`, with the `TimeLockServerTest` in the PR doing what you've mentioned here (though as we've discussed I should make this clearer).

Post offline discussion, we're probably not going to go ahead with an abstract test class here as the tests at this level seem to differ a fair bit depending on the implementation. The `verifyPortsAreAvailableOnStartUp()` would be a good (abstract!) integration test, though.